### PR TITLE
[Magiclysm] Adding appropriate Black Dragon mutations

### PR DESCRIPTION
**Pull request by Terrorforge**
#### Summary
Content "Add (versions of) some appropriate mutation to the Black Dragon category"

#### Purpose of change
The new Dense Bones mutation seemed appropriate for Black Dragons, so I wanted to add it. While poking around in the mutations, I noticed that while black dragons are carnivores, unlike other obligate carnivores they don't get the "Predator" line of mutations, nor do they get "Sapiovore". It seemed very weird to me that dragons of all things wouldn't be able to eat a princess, nor get the psychological benefits of being apex predators, so I decided to give them those traits as well.

However, PRED3 and PRED4 ("Predator" and "Apex Predator") have an associated intelligence debuff, which seemed inappropriate for a category that's all about magic, to the point where it already has several intelligence buffs, so I made alternate versions of those traits just for dragons. PRED3_DRAGON has no intelligence debuff, and PRED4_DRAGON even has a small intelligence buff (+1) because that seemed both funny and in keeping with the theme.

#### Describe the solution
Extended DENSE_BONES, PRED1, PRED2 and SAPIOVORE to include them in the black dragon category. Made alternate version of PRED3 and PRED4 without the int penalty.

#### Testing
Made an invincible character, took a boatload of mutagen and waited until i had every black dragon mutation to make sure the new ones got added.

Just as a point of interest, Black Dragon has so goddamn many mutations that this took two in-game weeks, and I had to reset my instability halfway through because I had like a 70% chance of bad mutations, but there were no more bad mutations to get.

#### Additional context
Since black dragons in Magiclysm are described as not merely predatory but actively malevolent, I want to give them Killer Drive as well. Unfortunately, Killer Drive conflicts with Sapiovore, and renders it largely irrelevant anyway. I could've given them Killer Drive instead, but it's a non-threshold mutation, which would conflict with the implication that Sapiovore represents crossing the moral event horizon where you're truly no longer human. I couldn't make a new trait, either, because the Killer Drive functionality is hardcoded. Might come back to it, but this is fine for now.

e: this change will of course also make it possible (and inevitable) for dragon mutants to become carnivores regardless of instability levels. This is intended; not only does it match the behavior of other predator categories, it also makes it a little harder to dodge the downsides of what's supposed to be a very "high highs, low lows" tree

### CODE REVIEW

File: data/mods/Magiclysm/mutations/mutations.json [hunk 1]
```diff
@@ -742,5 +742,67 @@
...
+    "id": "PRED2",
+    "copy-from": "PRED2",
+    "extend": { "category": [ "DRAGON_BLACK" ], "threshreq": [ "THRESH_DRAGON_BLACK" ] }
+  },
+  {
+    "type": "mutation",
+    "id": "PRED3_DRAGON",
+    "name": { "str": "Draconic Predator" },
+    "points": 4,
+    "description": "You have become something more than human. Something better - and far, far worse. Combat skills are easy to learn and maintain, and unlike lesser predators, you mental faculties don'r suffer at all.",
```
Comment (Maleclypse) [inline data/mods/Magiclysm/mutations/mutations.json:769]:
```suggestion
    "description": "You have become something more than human.  Something better - and far, far worse. Combat skills are easy to learn and maintain, and unlike lesser predators, you mental faculties don'r suffer at all.",
```

**Review by Maleclypse — APPROVED** [review]
This adds uniqueness to Black Dragon.  Good job!

File: data/mods/Magiclysm/mutations/mutations.json [hunk 2]
```diff
@@ -742,5 +742,67 @@
...
+    "id": "PRED2",
+    "copy-from": "PRED2",
+    "extend": { "category": [ "DRAGON_BLACK" ], "threshreq": [ "THRESH_DRAGON_BLACK" ] }
+  },
+  {
+    "type": "mutation",
+    "id": "PRED3_DRAGON",
+    "name": { "str": "Draconic Predator" },
+    "points": 4,
+    "description": "You have become something more than human.  Something better - and far, far worse. Combat skills are easy to learn and maintain, and unlike lesser predators, your mental faculties don't suffer at all.",
```
Comment (dseguin) [inline data/mods/Magiclysm/mutations/mutations.json:769]:
```suggestion
    "description": "You have become something more than human.  Something better - and far, far worse.  Combat skills are easy to learn and maintain, and unlike lesser predators, your mental faculties don't suffer at all.",
```

**Comment by adamkad1** [comment]
I think dragons can be omnivores but eh

**Comment by Terrorforge** [comment]
Magiclysm's black dragons are quite explicitly characterized as evil and murderous - and, crucially, were already carnivores. They just didn't get the post-threshold bonuses usually associated with carnivores for some reason.

If someone finally gets around to making other colors of dragon, as was originally the intention, I wouldn't mind seeing omnivorous or even herbivorous dragons. Or hell, dragons with even weirder diets, like a gold dragon that only eats metal or a blue dragon that only drinks water.

**Pull request merged**